field/composite_test.go

```diff
@@ -543,6 +543,80 @@ func TestCompositePackingWithTags(t *testing.T) {
 		require.Equal(t, "120102AB0202CD", string(packed))
 	})
 
+	t.Run("Pack correctly ignores excess subfields in excess of the length described by the prefix", func(t *testing.T) {
+		type ExcessSubfieldsTestData struct {
+			F1 *String
+			F2 *Numeric
+			F3 *String
+			F4 *String
+			F5 *String
+		}
+
+		excessSubfieldsSpec := &Spec{
+			Length:      5,
+			Description: "POS Data",
+			Pref:        prefix.EBCDIC1047.LL,
+			Tag: &TagSpec{
+				Sort: sort.StringsByInt,
+			},
+			Subfields: map[string]Field{
+				"1": NewString(&Spec{
+					Length:      1,
+					Description: "POS Terminal Attendance",
+					Enc:         encoding.EBCDIC1047,
+					Pref:        prefix.EBCDIC1047.Fixed,
+				}),
+				"2": NewNumeric(&Spec{
+					Length:      1,
+					Description: "Reserved for Future Use",
+					Enc:         encoding.EBCDIC1047,
+					Pref:        prefix.EBCDIC1047.Fixed,
+				}),
+				"3": NewString(&Spec{
+					Length:      1,
+					Description: "POS Terminal Location",
+					Enc:         encoding.EBCDIC1047,
+					Pref:        prefix.EBCDIC1047.Fixed,
+				}),
+				"4": NewString(&Spec{
+					Length:      1,
+					Description: "POS Cardholder Presence",
+					Enc:         encoding.EBCDIC1047,
+					Pref:        prefix.EBCDIC1047.Fixed,
+				}),
+				"5": NewString(&Spec{
+					Length:      1,
+					Description: "POS Card Presence",
+					Enc:         encoding.EBCDIC1047,
+					Pref:        prefix.EBCDIC1047.Fixed,
+				}),
+			},
+		}
+
+		data := &ExcessSubfieldsTestData{}
+
+		composite := NewComposite(excessSubfieldsSpec)
+		err := composite.SetData(data)
+		require.NoError(t, err)
+
+		// Subfield 4 & 5 falls outside of the bounds of the 3 byte limit imposed
```

Comment:
Also, I'm not sure I understand this bit of the PR. How does the early return help with fields that fall outside of the bounds? Also, where 3 byte limit prefix is? Is it because of `ebcdic1047`? Sorry for the dumb question.

Reply:
Hi, this probably wasn't explained the best by me - the motivation behind this PR was to allow  length-prefixed composites that with non-tagged subfields to be parsed only up to the specified length, so that non-populated subfields in the composite could be ignored without erroring/panicking.

The problem with the current implementation is when trying to parse a composite without tagged subfields, Composite loops through all the subfield IDs in the spec regardless and errors when it tried to unpack a zero-length data buffer for a subfield not included in the data.  

A possible fix would have been to just add check to Composite to stop parsing when we'd read >= the length of the data, but this caused an issue where errors weren't being accurately reported for invalid length data that was too long (i.e the spec only allows 4 bytes but 7 bytes were provided, it would stop reading after >= 4 bytes, so it wouldn't accurately highlight the issue with the combined length of the subfields). 

So now I attempted to get the desired result by focusing on not panicking when attempting to unpack zero-length data fields.

Reply:
In the original example, I was referring to the '03' prefix when mentioning the 3 byte limit (i.e. [03 | 102] -> [length prefix | data], but I've updated it the most recent commit to hopefully make it clearer